src/Bridge/Doctrine/MongoDB/CollectionDataProvider.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\Doctrine\MongoDB;
+
+use ApiPlatform\Core\Api\CollectionDataProviderInterface;
+use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ODM\MongoDB\DocumentRepository;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDB\Extension\QueryResultExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDB\Extension\QueryCollectionExtensionInterface;
+
+/**
+ * Collection data provider for the Doctrine ORM.
+ */
+class CollectionDataProvider implements CollectionDataProviderInterface
```

Comment:
Yeah, the interfaces for query extensions should be pulled out of ORM also i think. The paginator could be abstracted since the only read difference if the getResult and applyToCollection.

If you want me to proceed with this give me the go ahead :)